thrust/system/cuda/memory.h

```diff
@@ -64,66 +64,12 @@ inline __host__ __device__ pointer<T> malloc(std::size_t n);
  */
 inline __host__ __device__ void free(pointer<void> ptr);
 
-// XXX upon c++11
-// template<typename T>
-// using allocator = thrust::mr::stateless_resource_allocator<T, memory_resource>;
-
 /*! \p cuda::allocator is the default allocator used by the \p cuda system's containers such as
  *  <tt>cuda::vector</tt> if no user-specified allocator is provided. \p cuda::allocator allocates
  *  (deallocates) storage with \p cuda::malloc (\p cuda::free).
  */
-template <typename T>
-struct allocator
-    : thrust::mr::stateless_resource_allocator<
-        T,
-        system::cuda::memory_resource
-    >
-{
-private:
-    typedef thrust::mr::stateless_resource_allocator<
-        T,
-        system::cuda::memory_resource
-    > base;
-
-public:
-  /*! The \p rebind metafunction provides the type of an \p allocator
-   *  instantiated with another type.
-   *
-   *  \tparam U The other type to use for instantiation.
-   */
-  template <typename U>
-  struct rebind
-  {
-    /*! The typedef \p other gives the type of the rebound \p allocator.
-     */
-    typedef allocator<U> other;
-  };
-
-  /*! No-argument constructor has no effect.
-   */
-  __host__
-  inline allocator() {}
-
-  /*! Copy constructor has no effect.
-   */
-  __host__ __device__
-  inline allocator(const allocator & other) : base(other) {}
-
-  /*! Constructor from other \p allocator has no effect.
-   */
-  template <typename U>
-  __host__ __device__
-  inline allocator(const allocator<U> & other) : base(other) {}
-
-#if THRUST_CPP_DIALECT >= 2011
-  allocator & operator=(const allocator &) = default;
-#endif
-
-  /*! Destructor has no effect.
-   */
-  __host__ __device__
-  inline ~allocator() {}
-};    // struct allocator
+template<typename T>
+using allocator = thrust::mr::stateless_resource_allocator<T, memory_resource>;
```

Comment:
`memory_resource` needs to be qualified as `system::cuda::memory_resource` (see original)